### PR TITLE
Easier Data Access

### PR DESCRIPTION
**Pull request by iifeoluwa**
Implemented the key => value pair functionality to make the library more stimulating for use. I also modified the documentation to reflect the changes made.

**Before**
before

**After**
mvrd

I'd also appreciate if you could look at the code and suggest improvements/corrections.
Thanks @unicodeveloper!

### CODE REVIEW

File: src/Mvrd.php [hunk 1]
```diff
@@ -80,13 +80,40 @@ public function getData()
...
+                        $data[$string[0]] = $string[1] .' '. $string[2];
+                    }else{
+                        $string[0] = $string[0] == 'Isssue' ? 'Issue' : $string[0];
+                        $data[$string[0].$string[1]] =  $string[2];
+                    }                   
+                }else{
+                    $data[$string[0]] =  $string[1];
+                }
+                
+                return $data;
```
Comment (unicodeveloper) [inline src/Mvrd.php]: This whole part needs to be refactored. Right now it's very confusing for someone that needs to contribute to the code.

Also, it should be extracted into a function on its own and called in here

File: src/Mvrd.php [hunk 2]
```diff
@@ -80,13 +80,40 @@ public function getData()
...
-                return trim(str_replace(["\n", "  "], '', $node->text()));
+                $formData = trim(preg_replace('/\s+/', ' ', $node->text()));
+
+                // Prepare data to be stored as key=>value pairs
+                $string = explode(' ', $formData, 3);   
+                // Use the number of spaces in the string as a requisite to handle them.
+                // e.g Color field(1 space) contains less spaces than Plate Number(2 spaces) e.t.c
+                if (substr_count($formData, ' ') > 1) {
+                    // Check if the string holds Vehicle Model data as this needs to be stored differently.
+                    $string[0] == 'Model' ? $true = true : $true = false;
```
Comment (unicodeveloper) [inline src/Mvrd.php]: The use of variable $true for values of `true` and `false` is questionable. Imagine assigning hot water to ice water variable 😅.

File: src/Mvrd.php [hunk 3]
```diff
@@ -80,13 +80,40 @@ public function getData()
...
+                    if ($true) {
+                        $data[$string[0]] = $string[1] .' '. $string[2];
+                    }else{
+                        $string[0] = $string[0] == 'Isssue' ? 'Issue' : $string[0];
+                        $data[$string[0].$string[1]] =  $string[2];
+                    }                   
+                }else{
```
Comment (unicodeveloper) [inline src/Mvrd.php]: Identation is key. Makes your code readable

File: src/Mvrd.php [hunk 4]
```diff
@@ -80,13 +80,40 @@ public function getData()
...
+                $string = explode(' ', $formData, 3);   
+                // Use the number of spaces in the string as a requisite to handle them.
+                // e.g Color field(1 space) contains less spaces than Plate Number(2 spaces) e.t.c
+                if (substr_count($formData, ' ') > 1) {
+                    // Check if the string holds Vehicle Model data as this needs to be stored differently.
+                    $string[0] == 'Model' ? $true = true : $true = false;
+                    if ($true) {
+                        $data[$string[0]] = $string[1] .' '. $string[2];
+                    }else{
+                        $string[0] = $string[0] == 'Isssue' ? 'Issue' : $string[0];
```
Comment (unicodeveloper) [inline src/Mvrd.php]: A more readable version of this code will look like so:

```php

if (substr_count($formData, ' ') > 1) {
        $isModelPresent = ($string[0] === 'Model') ? true : false;

        if ($isModelPresent) {
            $data[$string[0]] = $string[1] .' '. $string[2];
        } else {
            $string[0] = ($string[0] === 'Isssue') ? 'Issue' : $string[0];
            $data[$string[0].$string[1]] =  $string[2];
        }
    } else {
        $data[$string[0]] =  $string[1];
    }
```

**Review by unicodeveloper — COMMENTED** [review]
Please apply the necessary changes as recommended. I will happy to take a look at it again when you are done.

File: src/Mvrd.php (hunk 1 above)
Comment (iifeoluwa) [inline src/Mvrd.php]: These comments are golden! I'll refactor and push updates.

Forgot to mention that I noticed 'Issue' was misspelt as 'Isssue' [lsmvaapvs ](http://www.lsmvaapvs.org/) so I changed it to 'Issue' at our end. What do you think about this?

Also, I'm sorry about the naming conflicts. I'm very poor at naming things.
Thanks a lot!

**Comment by iifeoluwa** [comment]
These comments are golden! I'll refactor and push updates.

Forgot to mention that I noticed 'Issue' was misspelt as 'Isssue' at [lsmvaapvs ](http://www.lsmvaapvs.org/) so I changed it to 'Issue' at our end. What do you think about this?

Also, I'm sorry about the naming conflicts. I'm very poor at naming things, working hard to improve.
Thanks a lot!

**Comment by iifeoluwa** [comment]
@unicodeveloper I have made modifications as suggested. Kindly review.

**Pull request merged**